plugins/modules/nd_setup.py

```diff
@@ -0,0 +1,355 @@
+            options=dict(
+                hostname=dict(type="str", required=True),
+                serial_number=dict(type="str", required=True),
+                management_ip_address=dict(type="str", required=True),
+                username=dict(type="str", required=True),
+                password=dict(type="str", required=True, no_log=True),
+                management_network=dict(type="dict", required=True, options=management_network_spec()),
+                data_network=dict(type="dict", required=True, options=data_network_spec()),
+                bgp=dict(type="dict", options=bgp_spec()),
+            ),
+        ),
+        state=dict(type="str", default="present", choices=["present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["cluster_name", "ntp_server", "dns_server", "app_network", "service_network", "nodes"]],
+        ],
+        required_together=[
+            ["proxy_username", "proxy_password"],
+        ],
+    )
+
+    nd = NDModule(module)
+
+    cluster_name = nd.params.get("cluster_name")
+    ntp_server = nd.params.get("ntp_server")
+    dns_server = nd.params.get("dns_server")
+    proxy_server = nd.params.get("proxy_server")
+    proxy_username = nd.params.get("proxy_username")
+    proxy_password = nd.params.get("proxy_password")
+    ignore_proxy = nd.params.get("ignore_proxy")
+    dns_search_domain = nd.params.get("dns_search_domain")
+    app_network = nd.params.get("app_network")
+    service_network = nd.params.get("service_network")
+    nodes = nd.params.get("nodes")
+    state = nd.params.get("state")
+
+    if state == "query":
+        nd.existing = nd.request("/clusterstatus/install", method="GET")
+    else:
+        if len(cluster_name) > 63 or len(re.findall(r"[^a-zA-Z0-9 -]|^-| *-$", cluster_name)) > 0:
+            nd.fail_json(
+                "A length of 1 to 63 characters is allowed and valid characters include letters, digits and hyphen. "
+                "The name cannot start or end with a hyphen."
+            )
+
+        payload = {
+            "clusterConfig": {
+                "name": cluster_name,
+                "ntpServers": ntp_server,
+                "nameServers": dns_server,
+                "searchDomains": dns_search_domain,
+                "ignoreHosts": ignore_proxy,
+                "proxyServers": [
+                    {
+                        "proxyURL": proxy_server,
+                        "username": proxy_username,
+                        "password": proxy_password,
+                    }
+                ],
+                "appNetwork": app_network,
+                "serviceNetwork": service_network,
+            },
+            "nodes": [
+                {
+                    "hostName": node.get("hostname"),
+                    "serialNumber": node.get("serial_number"),
+                    "dataNetwork": {
```

Comment:
why do we need the vlan option during input of data network? because I do not see it used in the payload

Reply:
I missed adding it here and it is actually an option that can be configured.